hadoop-hdds/interface-client/src/main/proto/hdds.proto

```diff
@@ -479,4 +479,19 @@ message DeletedBlocksTransactionInfo {
     optional int64 containerID = 2;
     repeated int64 localID = 3;
     optional int32 count = 4;
-}
\ No newline at end of file
+}
+
+message CompactionFileInfoProto {
+    required string fileName = 1;
+    optional string startKey = 2;
+    optional string endKey = 3;
+    optional string columnFamily = 4;
+}
+
+message CompactionLogEntryProto {
+    required uint64 dbSequenceNumber = 1;
```

Comment:
We should avoid using `required`, Please see here: https://protobuf.dev/programming-guides/proto2/
proto3 has already removed it. It is required for the application to handle this. Let me know what you think about it.

Reply:
I wasn't aware that `required` has been removed from proto3. The only reason I used `required` to show that a field is mandatory. I don't get why they think "It is nearly impossible to safely change a field from required to optional". To me, it should be OK to change required to optional but not vice-versa.

Since it is suggested that required fields should be implemented at the application layer instead, I will change it to optional.